### PR TITLE
Make sure kubelet gets restarted after package version change

### PR DESCRIPTION
**Pull request by danifr**
Fix https://github.com/puppetlabs/puppetlabs-kubernetes/issues/524

### CODE REVIEW

**Comment by danifr** [comment]
Hmm I am not sure why the tests failed.

 I am doing something similar to: https://github.com/puppetlabs/puppetlabs-kubernetes/blob/main/manifests/packages.pp#L357 (referencing a Service declared in the service.pp class)

I someone can give me a hint I'd appreciated.

**Comment by danifr** [comment]
Well, I don't know what I can do to make the tests pass... I tried a few different approaches but I always get an error. I wonder if the tests are OK.

Just so you know, same patch, running on my infra, works as expected (kubelet services gets restarted after a package update):
```
Notice: /Stage[main]/Kubernetes::Packages/Package[kubelet]/ensure: ensure changed '1.20.8-0' to '1.21.0' (corrective)
Info: /Stage[main]/Kubernetes::Packages/Package[kubelet]: Scheduling refresh of Service[kubelet]
Notice: /Stage[main]/Kubernetes::Packages/Package[kubectl]/ensure: ensure changed '1.20.8-0' to '1.21.0' (corrective)
Notice: /Stage[main]/Kubernetes::Packages/Package[kubeadm]/ensure: ensure changed '1.20.8-0' to '1.21.0' (corrective)
```
@daianamezdrea can you please review it? Thanks

**Comment by daianamezdrea** [comment]
Hi @danifr, thank you ! Yes, I'll have a look, I'll try to debug it locally and see what is not working fine

**Comment by daianamezdrea** [comment]
Hi @danifr, pushed some changes to see if the specs are going to pass

**Comment by danifr** [comment]
@daianamezdrea OMG thanks a lot!!!

**Comment by danifr** [comment]
Hello again, can we get this change merged and have it available within the next release? Thank you!

**Comment by danifr** [comment]
@daianamezdrea Hi! sorry for insisting, did you have some time to look at this? I think we can safely merge it into main

**Comment by daianamezdrea** [comment]
Hi @danifr, I required @carabasdaniel review on the changes I've made and he said that the dependencies added transform the unit tests in acceptance (with the changes I did) so I need to re-think the issue, sorry for the late response, looking now again how to approach it

**Comment by danifr** [comment]
Hi @daianamezdrea happy new year! is there something I can do to have this merged?

**Comment by danifr** [comment]
Is there something else I can do so we can get this one merged?

**Comment by chelnak** [comment]
Looks like there were some failing tests on the last run.

I've re kicked them so we can get a fresh view of what is happening.

**Comment by chelnak** [comment]
Looks like we have some conflicts. Could you rebase on current main please?

**Comment by danifr** [comment]
Hi @chelnak, sorry I missed you comment. I just rebased from main and the tests are still failing...

I don't understand why though... the only relevant change that this PR introduces is:

https://github.com/puppetlabs/puppetlabs-kubernetes/pull/525/files#diff-98b7734749457d0f7b07bfccf04f7316e31a48cdf3e02a46cb3cb8209daecb29R128

Maybe you know what's going on.

**Comment by deric** [comment]
@danifr For failing tests see #565

**Comment by danifr** [comment]
Still failing... I will try removing @daianamezdrea commit from this PR. Let's see

**Comment by danifr** [comment]
Blows my mind that is fails with: 
```
        # Puppet::Parser::Compiler::CatalogValidationError:
       #   Could not find resource 'Service[kubectl]' in parameter 'notify' (file: /home/runner/work/puppetlabs-kubernetes/puppetlabs-kubernetes/spec/fixtures/modules/kubernetes/manifests/packages.pp, line: 368)
```
Because of: https://github.com/puppetlabs/puppetlabs-kubernetes/pull/525/files#diff-7395f2f3aeb84af1465802a40aed5e8226f466887bccbf9a8e4599ea73f50964R386

But it does not say anything about: https://github.com/puppetlabs/puppetlabs-kubernetes/blob/main/manifests/packages.pp#L357
that is basically the same thing.

File: manifests/packages.pp [hunk 1]
```diff
@@ -359,29 +359,37 @@
...
-        owner   => 'root',
-        group   => 'root',
-        content => template('kubernetes/kubernetes_apt_package_pins.erb'),
+  if $create_repos {
+    case $facts['os']['family'] {
+      'Debian': {
+        package { $kube_packages:
+          ensure  => $kubernetes_package_version,
+          require => Class['apt::update'],
+          notify  => Service['kubectl'],
```
Comment (deric) [inline manifests/packages.pp:368]: @danifr There's no such service as `kubectl`. Try using `kubelet` instead :slightly_smiling_face:

**Comment by jordanbreen28** [comment]
@danifr looks like there are some merge conflicts. Could you resolve these so we can assess how to proceed with this.

**Comment by danifr** [comment]
I just did it. Thanks for the heads-up :)

**Comment by GSPatton** [comment]
hey @danifr. I pulled your PR and ran the tests locally. The tests fail on your branch but pass on the master branch so the failures are due to your changes.

The failing tests complain about missing k8s-related files e.g. `expected to contain File[/etc/systemd/system/kubelet.service.d/20-cloud.conf] with content =~ /--cloud-provider=openstack/` etc. This may be a good starting point.

Hopefully this helps!

**Comment by danifr** [comment]
@GSPatton sorry did not have time to work on this. I can open a new PR in the future if I managed to get it working.

I need to figure out how to run the test locally, so I can make sure the test pass before opening a PR. Do you have some docs or info on how to run them locally?

**Pull request closed without merging**